psydac/feec/tests/test_feec_conf_projectors_cart_2d.py

```diff
@@ -46,10 +45,10 @@ def get_polynomial_function(degree, hom_bc_axes, domain):
 @pytest.mark.parametrize('nc', [5])
 @pytest.mark.parametrize('reg', [0])
 @pytest.mark.parametrize('hom_bc', [False, True])
-@pytest.mark.parametrize('domain_name', ["4patch_nc", "2patch_nc"])
+@pytest.mark.parametrize('domain_name', ["1patch", "4patch_nc", "2patch_nc"])
 @pytest.mark.parametrize("nonconforming, full_mom_pres",
                          [(True, True), (False, True)])
-# NOTE (MCP march 2025): momentum conservation fails for nc = 4 and degree = 3, why?
+
```

Comment:
have you seen some failures regarding the previous comment?

Reply:
Yes, it does indeed fail in the non-conforming case, but also raises the warning: 

> "Careful, the correction term is currently not independent of the mesh."

because `space_1d.ncells <= p_moments + 1`. Iirc this is expected, because in the non-conforming case we calculate the correction coefficients once on the coarsest patch and use them for all spaces. If these corrections use DOF that lie too far in the interior, they get dependent on the resolution which is the problem here.

Reply:
What do you mean by "dependend on the resolution" ? The order of moment preservation should not only depend on the dimension of the patch space. If this dimension is large enough the requested order should be matched, otherwise an error (or a warning) should be thrown.